backend/clubs/permissions.py

```diff
@@ -81,6 +81,23 @@ def has_permission(self, request, view):
         else:
             return True
 
+    def has_object_permission(self, request, view, obj):
+        """
+        Do not allow transitions from and to club fair event.
+        """
+        FAIR_TYPE = 4
+        old_type = obj.type
+        new_type = request.data.get("type", old_type)
+
+        if view.action in ["update", "partial_update"]:
+            if old_type == FAIR_TYPE and new_type != FAIR_TYPE:
+                return False
+
+            if old_type != FAIR_TYPE and new_type == FAIR_TYPE:
```

Comment:
Is it possible to use XOR (`^`) here? This solution works fine and is very readable. I wonder if having the two values be integers would screw anything up.

Also, in general it is Pythonic to use `not a == b` instead of `a != b`.